### PR TITLE
ROX-18606: add s390x and ppc64le support to scanner ci builds

### PR DESCRIPTION
**Pull request by kcrane**
add s390x and ppc64le support to scanner ci builds

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -12,6 +12,38 @@ on:
...
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Define the matrix for build jobs
+      id: define-scanner-job-matrix
+      run: |
+        source './scripts/ci/lib.sh'
+
+        # If goarch is updated, be sure to update architectures in push-manifests below.
+        matrix='{ "build_and_push": { "name":["default"], "goos":["linux"], "goarch":["amd64", "arm64", "ppc64le", "s390x"] }, "push_manifests": { "name":["default"] } }'
```
Comment (RTann) [inline .github/workflows/ci.yaml:31]: no need for `"name"` since we don't do `prerelease` builds in this repo

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -12,6 +12,38 @@ on:
...
+
+        # Conditionally add a prerelease build (binaries built with GOTAGS=release)
+        if ! is_tagged; then
+          if ! is_in_PR_context || pr_has_label ci-build-prerelease; then
+            matrix="$(jq '.build_and_push.name += ["prerelease"]' <<< "$matrix")"
+            matrix="$(jq '.push_manifests.name += ["prerelease"]' <<< "$matrix")"
+          fi
+        fi
```
Comment (RTann) [inline .github/workflows/ci.yaml:39]: not needed, so you can delete this

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -12,6 +12,38 @@ on:
...
+
+        echo "Job matrix after conditionals:"
```
Comment (RTann) [inline .github/workflows/ci.yaml:41]: can remove this echo, but good to keep the next line

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -52,19 +88,25 @@ jobs:
 
       - uses: ./.github/actions/job-preamble
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
```
Comment (RTann) [inline .github/workflows/ci.yaml:96]: I don't think this is needed since it's just building the binaries

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -246,13 +300,19 @@ jobs:
...
```
Comment (RTann) [inline .github/workflows/ci.yaml:308]: I don't think this is needed

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -427,11 +497,6 @@ jobs:
       - name: diff-dumps
         run: ./scripts/ci/jobs/diff-dumps.sh
 
-      - uses: actions/upload-artifact@v4
-        with:
-          name: diff-dumps-inspect
-          path: /tmp/diff-dumps-inspect
```
Comment (RTann) [inline .github/workflows/ci.yaml:433]: how come this was removed?

**Review by RTann — COMMENTED** [review]
took any initial look. I noticed there are still ongoing commits, so I'm not sure if it's ready for review yet. Let me know when it's ready!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (kcrane) [reply .github/workflows/ci.yaml:31]: Yes, stil working on it.  Thanks for taking a quick look will add these changes on next checkin.  Do we need "push_manifests"?  don't see a reference to it in this repo either...
```
        matrix='{ "build_and_push": {"goos":["linux"], "goarch":["amd64", "arm64", "ppc64le", "s390x"] }, "push_manifests": { "name":["default"] } }'
```

File: .github/workflows/ci.yaml (hunk 6 above)
Comment (kcrane) [reply .github/workflows/ci.yaml:433]: good catch!  not sure what happened here...

**Comment by kcrane** [comment]
CI is stuck on expected results.  trying to close and open a new PR to fix.

**Pull request closed without merging**